source/regression1.md

```diff
@@ -862,20 +862,31 @@ could make or break whether or not they could afford put an offer on a house.
 
 Finally, {numref}`fig:07-predict-all` shows the predictions that our final
 model makes across the range of house sizes we might encounter in the
-Sacramento area&mdash;from 500 to 5000 square feet.  You have already seen a
+Sacramento area.
+Note that instead of predicting the house price only for those house sizes that happen to appear in our data,
+we predict it for evenly spaced values between the minimum and maximum in the data set
+(roughly 500 to 5000 square feet).
+We superimpose this prediction line on a scatter
+plot of the original housing price data,
+so that we can qualitatively assess if the model seems to fit the data well.
+You have already seen a
 few plots like this in this chapter, but here we also provide the code that
-generated it as a learning challenge.
+generated it as a learning opportunity.
 
 ```{code-cell} ipython3
 :tags: [remove-output]
 
-sacr_preds = pd.DataFrame({"sqft": np.arange(500, 5001, 10)})
-sacr_preds = sacr_preds.assign(
-                   predicted = sacr_fit.predict(sacr_preds)
+# Create a grid of evenly spaced values along the range of the sqft data
+sqft_prediction_grid = pd.DataFrame({
+    "sqft": np.arange(sacramento['sqft'].min(), sacramento['sqft'].max(), 10)
+})
+# Predict the price for each of the sqft values in the grid
+sacr_preds = sqft_prediction_grid.assign(
+    predicted = sacr_fit.predict(sqft_prediction_grid)
 )
 
-# the base plot: the training data scatter plot
-base_plot = alt.Chart(sacramento_train).mark_circle().encode(
+# Plot all the houses
+base_plot = alt.Chart(sacramento).mark_circle(opacity=0.4).encode(
```

Comment:
I changed to all the data here (and in the next chapter) instead of just the training data to match the text. However, we could consider bigger changes to this snippet and creating a prediction for each of the values in the test dataset and then plot only the test data. This is what we do in the worksheets, and it would fit thematically here in the textbook since this section is under the heading "Evaluating on the test set". I am not sure about the initial motivation for comparing against all the data here?

Reply:
@trevorcampbell  What do you think of this comment? Currently the textbook only shows how to create predictions using the manual grid and then compare it with the all the data. In the assignments, we ask students to create predictions from the test data and plot it over the test data only.

Reply:
Let's just open an issue for that. I'm reluctant to mess with this right now too much -- when we originally wrote the book I remember this being a notable point of discussion, and in order to properly address this I'd need to go back and refresh my memory on all the various arguments surrounding.